### PR TITLE
fix(caldav): loop through all events for busy events

### PR DESCRIPTION
**Pull request by miaulalala**
## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Review by st3iny — APPROVED** [review]

File: apps/dav/lib/CalDAV/CalendarObject.php [hunk 1]
```diff
@@ -97,19 +97,16 @@ protected function isShared() {
...
-		if (isset($vObject->VTODO)) {
-			$vElement = $vObject->VTODO;
-		}
-		if (!is_null($vElement)) {
+		$vElements = array_filter($vObject->getComponents(), static function ($vElement) {
+			return $vElement instanceof Component\VEvent || $vElement instanceof Component\VJournal || $vElement instanceof Component\VTodo;
+		});
+
+		foreach ($vElements as $vElement) {
+			if (empty($vElement->select('SUMMARY'))) {
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/CalendarObject.php:107]: You could `trim()` before checking.

**Pull request merged**